File: data/src/main/java/com/tonitealive/app/data/GsonJsonSerializer.java
package com.tonitealive.app.data;

import com.google.gson.Gson;

import javax.inject.Singleton;

@Singleton
public class GsonJsonSerializer implements JsonSerializer {

    private final Gson gson;

    public GsonJsonSerializer(Gson gson) {
        this.gson = gson;
    }

    @Override
    public <T> String toString(T obj) {
        return gson.toJson(obj);
    }

    @Override
    public <T> T fromString(String s, Class<T> clazz) {
        return gson.fromJson(s, clazz);
    }
}

File: data/src/main/kotlin/com/tonitealive/app/data/GsonJsonSerializer.kt
package com.tonitealive.app.data

import com.google.gson.Gson
import javax.inject.Singleton

@Singleton
class GsonJsonSerializer(private val gson: Gson) : JsonSerializer {
    override fun <T> toString(obj: T): String {
        return gson.toJson(obj)
    }

    override fun <T> fromString(s: String, clazz: Class<T>): T {
        return gson.fromJson(s, clazz)
    }
}